test/Unit/Task/SecurityCheckerComposerauditTest.php

```diff
@@ -0,0 +1,168 @@
+            false,
+            $this->mockContext()
+        ];
+    }
+
+    public function provideFailsOnStuff(): iterable
+    {
+        yield 'exitCode1' => [
+            [],
+            $this->mockContext(RunContext::class, ['composer.lock']),
+            function () {
+                $this->mockProcessBuilder('composer', $process = $this->mockProcess(1));
+                $this->formatter->format($process)->willReturn('nope');
+            },
+            'nope'
+        ];
+    }
+
+    public function providePassesOnStuff(): iterable
+    {
+        yield 'exitCode0' => [
+            [],
+            $this->mockContext(RunContext::class, ['composer.lock']),
+            function () {
+                $this->mockProcessBuilder('composer', $this->mockProcess(0));
+            }
+        ];
+        yield 'exitCode0WhenRunAlways' => [
+            [
+                'run_always' => true
+            ],
+            $this->mockContext(RunContext::class, ['notrelated.php']),
+            function () {
+                $this->mockProcessBuilder('composer', $this->mockProcess(0));
+            }
+        ];
+    }
+
+    public function provideSkipsOnStuff(): iterable
+    {
+        yield 'no-files' => [
+            [],
+            $this->mockContext(RunContext::class),
+            function () {}
+        ];
+        yield 'no-composer-file' => [
+            [],
+            $this->mockContext(RunContext::class, ['thisisnotacomposerfile.lock']),
+            function () {}
+        ];
+    }
+
+    public function provideExternalTaskRuns(): iterable
+    {
+        yield 'defaults' => [
+            [],
+            $this->mockContext(RunContext::class, ['composer.lock']),
+            'composer',
+            [
+                'audit',
+                '--locked',
+                '--working-dir=./',
+            ]
+        ];
+
+        yield 'format' => [
+            [
+                'format' => 'json',
+            ],
+            $this->mockContext(RunContext::class, ['composer.lock']),
+            'composer',
+            [
+                'audit',
+                '--format=json',
+                '--locked',
+                '--working-dir=./',
+            ]
+        ];
+
+        yield 'locked' => [
+            [
+                'locked' => false,
+            ],
+            $this->mockContext(RunContext::class, ['composer.lock']),
+            'composer',
+            [
+                'audit',
+                '--working-dir=./',
+            ]
+        ];
+
+        yield 'no-dev' => [
+            [
+                'no_dev' => true,
+            ],
+            $this->mockContext(RunContext::class, ['composer.lock']),
+            'composer',
+            [
+                'audit',
+                '--locked',
+                '--no-dev',
+                '--working-dir=./',
+            ]
+        ];
+
+        yield 'working-dir' => [
+            [
+                'working_dir' => './',
```

Comment:
`./` Is the default value, can you test with `src` for example?

Reply:
Actually might need help here. When manually testing to another path the option works. But when modifying the test to another path I wasn't able to get the test to pass yet.

Also a side note, I think it might be a good idea for this to be fully optional ( default to null) but would require more logic to check the path for a changed composer.lock file.

Reply:
Thanks for bringing it up.

Reply:
You can probably take a look at https://github.com/phpro/grumphp/blob/9fb0b10469da75a0ebb0f3832edc9387ae45b0bb/src/Task/ComposerScript.php#L30

Reply:
I believe this is addressed now.